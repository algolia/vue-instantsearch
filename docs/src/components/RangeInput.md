---
title: RangeInput
mainTitle: Components
layout: main.pug
category: Components
withHeadings: true
navWeight: 6
editable: true
githubSource: docs/src/components/RangeInput.md
---

The range input widget allows a user to select a numeric range using a minimum and maximum input.

<a class="btn btn-static-theme" href="stories/?selectedKind=RangeInput">ðŸ•¹ try out live</a>

## Usage

```html
<ais-range-input :attribute="attribute"></ais-range-input>
```

## Props

Name | Type | Default | Description | Required
---|---|---|---|---
attribute | String | `defaultValue` | The name of the attribute in your record | yes
min | Number | - | Minimum value of the range | no
max | Number | - | Maximum value of the range | no
precision | Number | 0 | Number of digits after the decimal point to enforce | no
classNames | Object | | Override class names | no

## CSS classes

Here's a list of CSS classes exposed by this widget. To better understand the underlying
DOM structure, have a look at the generated DOM in your browser.

Note that you can pass the prop `class-names`, with an object of class names and their replacement to override this.

Class name | Description
---|---
`ais-RangeInput` | Container class
`ais-RangeInput--noRefinement` | Class added on the container when there are no refinements possible
`ais-RangeInput-form` | The form wrapper around the inputs and the submit button
`ais-RangeInput-separator` | The separator between the min and the max
`ais-RangeInput-button` | The button that triggers the submission of the form submission of the form
`ais-RangeInput-label` | Enclosing label of an input
`ais-RangeInput-input` | An input
`ais-RangeInput-input--min` | The minimum bound of the range
`ais-RangeInput-input--max` | The maximum bound of the range

classNames | Object | | Override class names | no

## Slots

Name | Scope | Description
---|---|---
default | `{ refine: (min, max) => void, currentRefinements: [number, number], noRefinements, range: {min: number, max: number}}` | Slot to override the DOM output. `refine` takes the updated values and then trigger the search. The `currentRefinements` contains the min and the max as set in the search state (and should be used to render the input values).
minLabel | | Slot for giving the min input a label
maxLabel | | Slot for giving the max input a label
separator| | Slot for modifying the separator between the two inputs
submitLabel | | Slot for modifying the label on the submit button
