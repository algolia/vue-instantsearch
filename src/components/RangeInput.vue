<template>
  <div :class="bem()">
    <slot name="header"></slot>

    <form @submit.prevent="onSubmit(refinement)">
      <input
        type="number"
        :class="bem('input', 'from')"
        :min="range.min"
        :max="range.max"
        :step="step"
        :placeholder="rangeForRendering.min"
        :value="refinementForRendering.min"
        @input="refinement.min = $event.target.value"
      />

      <slot name="separator">
        <span :class="bem('separator')">
          to
        </span>
      </slot>

      <input
        type="number"
        :class="bem('input', 'to')"
        :min="range.min"
        :max="range.max"
        :step="step"
        :placeholder="rangeForRendering.max"
        :value="refinementForRendering.max"
        @input="refinement.max = $event.target.value"
      />
      <slot name="submit">
        <button :class="bem('submit')">Ok</button>
      </slot>
    </form>

    <slot name="footer"></slot>
  </div>
</template>

<script>
import algoliaComponent from '../component';
import { FACET_OR } from '../store';

export default {
  mixins: [algoliaComponent],
  props: {
    attributeName: {
      type: String,
      required: true,
    },
    min: {
      type: Number,
    },
    max: {
      type: Number,
    },
    defaultRefinement: {
      type: Object,
      default() {
        return {};
      },
    },
    precision: {
      type: Number,
      default: 0,
      validator(value) {
        return value >= 0;
      },
    },
  },

  data() {
    return {
      blockClassName: 'ais-range-input',
    };
  },

  created() {
    const { min: minValue, max: maxValue } = this.defaultRefinement;

    let min;
    if (minValue !== undefined) {
      min = minValue;
    } else if (this.min !== undefined) {
      min = this.min;
    }

    let max;
    if (maxValue !== undefined) {
      max = maxValue;
    } else if (this.max !== undefined) {
      max = this.max;
    }

    this.searchStore.stop();

    this.searchStore.addFacet(this.attributeName, FACET_OR);

    if (min !== undefined) {
      this.searchStore.addNumericRefinement(this.attributeName, '>=', min);
    }

    if (max !== undefined) {
      this.searchStore.addNumericRefinement(this.attributeName, '<=', max);
    }

    this.searchStore.start();
    this.searchStore.refresh();
  },

  destroyed() {
    this.searchStore.stop();
    this.searchStore.removeFacet(this.attributeName);
    this.searchStore.start();
  },

  computed: {
    step() {
      return 1 / Math.pow(10, this.precision);
    },

    refinement() {
      const { numericValue: min } =
        this.searchStore.activeRefinements.find(
          ({ attributeName, type, operator }) =>
            attributeName === this.attributeName &&
            type === 'numeric' &&
            operator === '>='
        ) || {};

      const { numericValue: max } =
        this.searchStore.activeRefinements.find(
          ({ attributeName, type, operator }) =>
            attributeName === this.attributeName &&
            type === 'numeric' &&
            operator === '<='
        ) || {};

      return {
        min,
        max,
      };
    },

    range() {
      const { min: minRange, max: maxRange } = this;
      const { min: minStat, max: maxStat } = this.searchStore.getFacetStats(
        this.attributeName
      );

      const pow = Math.pow(10, this.precision);

      let min;
      if (minRange !== undefined) {
        min = minRange;
      } else if (minStat !== undefined) {
        min = minStat;
      } else {
        min = -Infinity;
      }

      let max;
      if (maxRange !== undefined) {
        max = maxRange;
      } else if (maxStat !== undefined) {
        max = maxStat;
      } else {
        max = Infinity;
      }

      return {
        min: min !== -Infinity ? Math.floor(min * pow) / pow : min,
        max: max !== Infinity ? Math.ceil(max * pow) / pow : max,
      };
    },

    rangeForRendering() {
      const { min, max } = this.range;

      const isMinInfinity = min === -Infinity;
      const isMaxInfinity = max === Infinity;

      return {
        min: !isMinInfinity && !isMaxInfinity ? min : '',
        max: !isMinInfinity && !isMaxInfinity ? max : '',
      };
    },

    refinementForRendering() {
      const { min: minValue, max: maxValue } = this.refinement;
      const { min: minRange, max: maxRange } = this.range;

      return {
        min: minValue !== undefined && minValue !== minRange ? minValue : '',
        max: maxValue !== undefined && maxValue !== maxRange ? maxValue : '',
      };
    },
  },

  methods: {
    nextValueForRefinment(hasBound, isReset, range, value) {
      let next;
      if (!hasBound && range === value) {
        next = undefined;
      } else if (hasBound && isReset) {
        next = range;
      } else {
        next = value;
      }

      return next;
    },

    onSubmit({ min: minNext = '', max: maxNext = '' }) {
      const { min: minRange, max: maxRange } = this.range;

      const hasMinBound = this.min !== undefined;
      const hasMaxBound = this.max !== undefined;

      const isMinReset = minNext === '';
      const isMaxReset = maxNext === '';

      const minNextAsNumber = !isMinReset ? parseFloat(minNext) : undefined;
      const maxNextAsNumber = !isMaxReset ? parseFloat(maxNext) : undefined;

      const newMinNext = this.nextValueForRefinment(
        hasMinBound,
        isMinReset,
        minRange,
        minNextAsNumber
      );

      const newMaxNext = this.nextValueForRefinment(
        hasMaxBound,
        isMaxReset,
        maxRange,
        maxNextAsNumber
      );

      this.searchStore.stop();

      this.searchStore.removeNumericRefinement(this.attributeName, '>=');
      if (newMinNext !== undefined) {
        this.searchStore.addNumericRefinement(
          this.attributeName,
          '>=',
          newMinNext
        );
      }

      this.searchStore.removeNumericRefinement(this.attributeName, '<=');
      if (newMaxNext !== undefined) {
        this.searchStore.addNumericRefinement(
          this.attributeName,
          '<=',
          newMaxNext
        );
      }

      this.searchStore.start();
      this.searchStore.refresh();
    },
  },
};
</script>
