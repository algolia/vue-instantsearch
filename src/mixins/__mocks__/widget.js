let state = {};

// we need to have state given by `component` before it is mounted, otherwise
// we can't render it in most cases (items, hits, etc. are used in the template)
// so we share a (mock) global state during a whole test.
//
// (a mock is imported once per test file, so the state is isolated between tests)
//
// This allows us to import this `__setState` function and call it in the test
// to give the necessary data before mounting.
export function __setState(newState) {
  state = newState;
}

export const createWidgetMixin = () => ({
  data() {
    return {
      state,
    };
  },
});
