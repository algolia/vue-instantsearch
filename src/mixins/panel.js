import Vue from 'vue';

export const PANEL_EMITTER_NAMESPACE = 'instantSearchPanelEmitter';
export const PANEL_CHANGE_EVENT = 'PANEL_CHANGE_EVENT';

export const createPanelProviderMixin = () => ({
  props: {
    emitter: {
      type: Object,
      required: false,
      default() {
        return new Vue({
          name: 'PanelProvider',
        });
      },
    },
  },
  provide() {
    return {
      [PANEL_EMITTER_NAMESPACE]: this.emitter,
    };
  },
  data() {
    return {
      canRefine: true,
    };
  },
  created() {
    this.emitter.$on(PANEL_CHANGE_EVENT, value => {
      this.updateCanRefine(value);
    });
  },
  beforeDestroy() {
    this.emitter.$destroy();
  },
  methods: {
    updateCanRefine(value) {
      this.canRefine = value;
    },
  },
});

export const createPanelConsumerMixin = ({ mapStateToCanRefine }) => ({
  inject: {
    emitter: {
      from: PANEL_EMITTER_NAMESPACE,
      default() {
        return {
          $emit: () => {},
        };
      },
    },
  },
  data() {
    return {
      state: null,
      hasAlreadyEmitted: false,
    };
  },
  watch: {
    state: {
      immediate: true,
      handler(nextState, previousState) {
        if (!nextState) {
          return;
        }

        const previousCanRefine = mapStateToCanRefine(previousState || {});
        const nextCanRefine = mapStateToCanRefine(nextState);

        if (!this.hasAlreadyEmitted || previousCanRefine !== nextCanRefine) {
          this.emitter.$emit(PANEL_CHANGE_EVENT, nextCanRefine);

          this.hasAlreadyEmitted = true;
        }
      },
    },
  },
});
