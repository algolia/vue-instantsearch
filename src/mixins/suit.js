import suit from '../util/suit';

export const createSuitMixin = ({ name }) => ({
  props: {
    classNames: {
      type: Object,
      default: undefined,
    },
  },
  methods: {
    suit(element, modifier) {
      const className = suit(name, element, modifier);
      const userClassName = this.classNames && this.classNames[className];
      if (userClassName) {
        return [className, userClassName].join(' ');
      }
      return className;
    },
  },
});
