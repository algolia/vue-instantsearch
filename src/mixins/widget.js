const defaultShouldUpdate = (previous, next) => previous !== next;

export const createWidgetMixin = ({ connect }) => ({
  inject: {
    instantSearchInstance: {
      name: 'instantSearchInstance',
      default() {
        const tag = this.$options._componentTag;
        throw new TypeError(
          `It looks like you forgot to wrap your Algolia search component
          "<${tag}>" inside of an "<ais-index>" component.`
        );
      },
    },
  },
  data() {
    return {
      state: null,
    };
  },
  computed: {
    widgetParams() {
      const { widgetParams } = this.$options;

      return Object.keys(widgetParams).reduce(
        (acc, name) => ({
          ...acc,
          [name]: widgetParams[name].getValue(this),
        }),
        {}
      );
    },
  },
  watch: {
    widgetParams(next, previous) {
      const { widgetParams } = this.$options;

      const shouldUpdateWidget = Object.keys(widgetParams).some(name => {
        const { shouldUpdate = defaultShouldUpdate } = widgetParams[name];

        return shouldUpdate(previous[name], next[name]);
      });

      if (shouldUpdateWidget) {
        this.instantSearchInstance.removeWidget(this.widget);
        this.widget = this.widgetFactory(next);
        this.instantSearchInstance.addWidget(this.widget);
      }
    },
  },
  created() {
    this.widgetFactory = connect(this.updateData, () => {});
    this.widget = this.widgetFactory(this.widgetParams);
    this.instantSearchInstance.addWidget(this.widget);
  },
  beforeDestroy() {
    this.instantSearchInstance.removeWidget(this.widget);
  },
  methods: {
    updateData(state = {}) {
      this.state = state;
    },
  },
});
