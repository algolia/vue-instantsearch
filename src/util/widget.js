export const addWidget = (connector, parent, widgetParams, render) => {
  const factory = connector(render);
  const widget = factory(widgetParams);
  parent.addWidgets([widget]);

  return widget;
};

export const forceRender = (widget, parent, instantSearchInstance) => {
  if (
    instantSearchInstance.__initialSearchResults &&
    !instantSearchInstance.started
  ) {
    if (typeof instantSearchInstance.__forceRender !== 'function') {
      throw new Error(
        'You are using server side rendering with <ais-instant-search> instead of <ais-instant-search-ssr>.'
      );
    }
    instantSearchInstance.__forceRender(widget, parent);
  }
};

export const removeWidget = (widget, parent) => {
  if (widget) {
    parent.removeWidgets([widget]);
  }
};
