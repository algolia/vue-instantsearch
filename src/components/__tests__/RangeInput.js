import { mount } from '@vue/test-utils';
import { __setState } from '../../mixins/widget';
import RangeInput from '../RangeInput.vue';

jest.mock('../../mixins/widget');
jest.mock('../../mixins/panel');

const defaultRange = {
  min: 0,
  max: 1000,
};

const defaultState = {
  start: [0, 1000],
  range: defaultRange,
  refine: () => {},
};

const defaultProps = {
  attribute: 'price',
};

it('accepts an attribute prop', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(RangeInput, {
    propsData: {
      ...defaultProps,
    },
  });

  expect(wrapper.vm.widgetParams.attribute).toBe('price');
});

it('accepts a min prop', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(RangeInput, {
    propsData: {
      ...defaultProps,
      min: 10,
    },
  });

  expect(wrapper.vm.widgetParams.min).toBe(10);
});

it('accepts a max prop', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(RangeInput, {
    propsData: {
      ...defaultProps,
      max: 500,
    },
  });

  expect(wrapper.vm.widgetParams.max).toBe(500);
});

it('accepts a precision prop', () => {
  __setState({
    ...defaultState,
  });

  const wrapper = mount(RangeInput, {
    propsData: {
      ...defaultProps,
      precision: 3,
    },
  });

  expect(wrapper.vm.widgetParams.precision).toBe(3);
});

describe('rendering', () => {
  it('displays correctly with default', () => {
    __setState({
      ...defaultState,
    });

    const wrapper = mount(RangeInput, {
      propsData: defaultProps,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('displays correctly with a min', () => {
    __setState({
      ...defaultState,
      range: {
        ...defaultRange,
        min: 100,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('displays correctly with a max', () => {
    __setState({
      ...defaultState,
      range: {
        ...defaultRange,
        max: 100,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('displays correctly with a min and a max', () => {
    __setState({
      ...defaultState,
      range: {
        min: 10,
        max: 37,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('displays correctly with a min refinement', () => {
    __setState({
      ...defaultState,
      start: [10, 500],
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.find('.ais-RangeInput-input--min').element.value).toBe('10');
  });

  it('displays correctly with a min refinement equal -Infinity', () => {
    __setState({
      ...defaultState,
      start: [-Infinity, 500],
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.find('.ais-RangeInput-input--min').element.value).toBe('');
  });

  it('displays correctly with a min refinement equal to min range', () => {
    __setState({
      ...defaultState,
      start: [10, 500],
      range: {
        ...defaultRange,
        min: 10,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.find('.ais-RangeInput-input--min').element.value).toBe('');
  });

  it('displays correctly with a max refinement', () => {
    __setState({
      ...defaultState,
      start: [10, 500],
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.find('.ais-RangeInput-input--max').element.value).toBe(
      '500'
    );
  });

  it('displays correctly with a max refinement equal Infinity', () => {
    __setState({
      ...defaultState,
      start: [10, Infinity],
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.find('.ais-RangeInput-input--max').element.value).toBe('');
  });

  it('displays correctly with a max refinement equal to max range', () => {
    __setState({
      ...defaultState,
      start: [10, 500],
      range: {
        ...defaultRange,
        max: 500,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    expect(wrapper.find('.ais-RangeInput-input--max').element.value).toBe('');
  });

  it('calls the Panel mixin with `range`', () => {
    __setState({
      ...defaultState,
      range: {
        min: 0,
        max: 10,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        attribute: 'price',
      },
    });

    const mapStateToCanRefine = () =>
      wrapper.vm.mapStateToCanRefine(wrapper.vm.state);

    expect(mapStateToCanRefine()).toBe(true);

    wrapper.setData({
      state: {
        range: {
          min: 0,
          max: 0,
        },
      },
    });

    expect(mapStateToCanRefine()).toBe(false);
  });
});

describe('refinement', () => {
  it('uses the value of the inputs when the form is submited', () => {
    const refine = jest.fn();

    __setState({
      ...defaultState,
      refine,
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        attribute: 'price',
      },
    });

    const minInput = wrapper.find('.ais-RangeInput-input--min');
    minInput.element.value = 100;
    minInput.trigger('change');

    const maxInput = wrapper.find('.ais-RangeInput-input--max');
    maxInput.element.value = 106;
    maxInput.trigger('change');

    const form = wrapper.find('form');
    form.trigger('submit');

    expect(refine).toHaveBeenLastCalledWith(['100', '106']);
  });

  it('refines correctly when `start` given and user clicks submit without changing input field', () => {
    const refine = jest.fn();
    __setState({
      refine,
      start: [50, 100],
      range: {
        min: 1,
        max: 5000,
      },
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    const form = wrapper.find('form');
    form.trigger('submit');

    expect(refine).toHaveBeenCalledTimes(1);
    expect(refine).toHaveBeenCalledWith([50, 100]);
  });

  it('refines correctly even when state changes', () => {
    const refine = jest.fn();
    __setState({
      ...defaultState,
      refine,
    });

    const wrapper = mount(RangeInput, {
      propsData: {
        ...defaultProps,
      },
    });

    // refine for the first time
    const minInput = wrapper.find('.ais-RangeInput-input--min');
    minInput.element.value = 10;
    minInput.trigger('change');

    const maxInput = wrapper.find('.ais-RangeInput-input--max');
    maxInput.element.value = 100;
    maxInput.trigger('change');

    const form = wrapper.find('form');
    form.trigger('submit');

    expect(refine).toHaveBeenCalledTimes(1);
    expect(refine).toHaveBeenCalledWith(['10', '100']);

    // update the state
    wrapper.setData({
      state: { start: [50, 200] }, // min: 10 -> 50, max: 100 -> 200
    });

    form.trigger('submit');
    expect(refine).toHaveBeenCalledTimes(2);
    expect(refine).toHaveBeenCalledWith([50, 200]);
  });
});
