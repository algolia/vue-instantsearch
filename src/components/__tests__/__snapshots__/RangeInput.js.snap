// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering displays correctly with a max 1`] = `

<div class="ais-RangeInput">
  <form class="ais-RangeInput-form">
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="0"
             max="100"
             placeholder="0"
             class="ais-RangeInput-input ais-RangeInput-input--min"
      >
    </label>
    <span class="ais-RangeInput-separator">
      to
    </span>
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="0"
             max="100"
             placeholder="100"
             class="ais-RangeInput-input ais-RangeInput-input--max"
      >
    </label>
    <button type="submit"
            class="ais-RangeInput-submit"
    >
      Go
    </button>
  </form>
</div>

`;

exports[`rendering displays correctly with a min 1`] = `

<div class="ais-RangeInput">
  <form class="ais-RangeInput-form">
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="100"
             max="1000"
             placeholder="100"
             class="ais-RangeInput-input ais-RangeInput-input--min"
      >
    </label>
    <span class="ais-RangeInput-separator">
      to
    </span>
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="100"
             max="1000"
             placeholder="1000"
             class="ais-RangeInput-input ais-RangeInput-input--max"
      >
    </label>
    <button type="submit"
            class="ais-RangeInput-submit"
    >
      Go
    </button>
  </form>
</div>

`;

exports[`rendering displays correctly with a min and a max 1`] = `

<div class="ais-RangeInput">
  <form class="ais-RangeInput-form">
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="10"
             max="37"
             placeholder="10"
             class="ais-RangeInput-input ais-RangeInput-input--min"
      >
    </label>
    <span class="ais-RangeInput-separator">
      to
    </span>
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="10"
             max="37"
             placeholder="37"
             class="ais-RangeInput-input ais-RangeInput-input--max"
      >
    </label>
    <button type="submit"
            class="ais-RangeInput-submit"
    >
      Go
    </button>
  </form>
</div>

`;

exports[`rendering displays correctly with default 1`] = `

<div class="ais-RangeInput">
  <form class="ais-RangeInput-form">
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="0"
             max="1000"
             placeholder="0"
             class="ais-RangeInput-input ais-RangeInput-input--min"
      >
    </label>
    <span class="ais-RangeInput-separator">
      to
    </span>
    <label class="ais-RangeInput-label">
      <input type="number"
             step="1"
             min="0"
             max="1000"
             placeholder="1000"
             class="ais-RangeInput-input ais-RangeInput-input--max"
      >
    </label>
    <button type="submit"
            class="ais-RangeInput-submit"
    >
      Go
    </button>
  </form>
</div>

`;
